goal_src/jak2/engine/common_objs/conveyor.gc

```diff
@@ -7,3 +7,492 @@
+       and return it as well.  Otherwise, set it to `0`"
+  (let ((actor-options (res-lump-value (-> obj entity) 'options actor-option :time -1000000000.0)))
+    (the-as
+      ambient-sound
+      (cond
+        ((not (logtest? (actor-option no-amb-sound) actor-options))
+         (let ((sound
+                 (the-as object (new 'process 'ambient-sound (static-sound-spec "conveyor" :fo-max 80) (-> obj root trans)))
+                 )
+               )
+           (set! (-> obj sound) (the-as ambient-sound sound))
+           sound
+           )
+         )
+        (else
+          (set! (-> obj sound) (the-as ambient-sound 0))
+          0
+          )
+        )
+      )
+    )
+  )
+
+(defmethod conveyor-method-26 conveyor ((obj conveyor) (proc-focus process-focusable))
+  "TODO - conveyor section related, perhaps related to moving the process along the belt?"
+  (let ((vec (new 'stack-no-clear 'vector)))
+    (set! (-> vec quad) (-> (get-trans proc-focus 0) quad))
+    (set! (-> vec w) 1.0)
+    (when (>= (vector4-dot vec (the-as vector (-> obj leading))) 0.0)
+      (let* ((sections (-> obj sections))
+             (section-count (-> sections length))
+             )
+        (dotimes (section-idx section-count)
+          (let ((section (-> sections data section-idx)))
+            (when (< (vector4-dot vec (the-as vector (-> section trailing))) 0.0)
+              (let ((vec-temp (new 'stack-no-clear 'vector)))
+                (vector-! vec-temp vec (-> section start))
+                (when (>= (-> obj belt-radius) (fabs (vector-dot vec-temp (-> section radial-dir))))
+                  (let* ((f0-7 (vector-dot vec-temp (-> section pull-dir)))
+                         (f1-6 (- (-> vec-temp y) (* (-> section pull-dir y) f0-7)))
+                         )
+                    (when (>= (-> obj pull-y-threshold) (fabs f1-6))
+                      (let ((a2-8 (new 'stack-no-clear 'vector)))
+                        (let ((f0-10 (-> obj speed)))
+                          (if (< section-idx (-> obj speed-mult-array-len))
+                              (set! f0-10
+                                    (* f0-10 (-> (the-as (pointer float) (+ (the-as uint (-> obj speed-mult-array)) (* section-idx 4)))))
+                                    )
+                              )
+                          (vector-float*! a2-8 (-> section pull-dir) (* f0-10 (-> self clock seconds-per-frame)))
+                          )
+                        (send-event proc-focus 'push-trans a2-8 3000)
+                        )
+                      )
+                    )
+                  )
+                )
+              (return #f)
+              )
+            )
+          )
+        )
+      #f
+      )
+    )
+  )
+
+(defmethod conveyor-method-27 conveyor ((obj conveyor))
+  "TODO - collision related, has some dead code as well (previous iteration?)"
```

Comment:
This is super interesting.
This code has a `(when #t` and a `(when #f` case. The `(when #t` uses Jak 2's new collision system, the `(when #f` looks like it uses jak 1's style.